ttnn/CMakeLists.txt

```diff
@@ -678,10 +803,25 @@ if(CMAKE_CXX_COMPILER_ID STREQUAL "GNU")
 endif()
 
 target_include_directories(ttnn PUBLIC ${TTNN_PUBLIC_INCLUDE_DIRS})
-target_link_libraries(ttnn PUBLIC ${TTNN_PUBLIC_LINK_LIBRARIES})
+target_link_libraries(
+    ttnn
+    PUBLIC
+        ${TTNN_PUBLIC_LINK_LIBRARIES}
+        ${ALL_TTNN_SUBLIBRARIES}
+        ttnn_tensor
+        ttnn_ccl
+        ttnn_ccl_exp
+)
 target_link_directories(ttnn PUBLIC ${TTNN_PUBLIC_LINK_DIRS})
-target_precompile_headers(ttnn PRIVATE ${TTNN_PRECOMPILED_HEADERS})
-TT_ENABLE_UNITY_BUILD(ttnn)
+target_precompile_headers(ttnn REUSE_FROM ${PRECOMPILED_HEADER_TARGET})
+set_target_properties(
+    ttnn
+    PROPERTIES
+        DEFINE_SYMBOL
+            ""
+)
```

Comment:
whats the point of this code?

Reply:
This is necessary to make the pre-compiled headers work. I create the PCH only once and reuse them for all the sub libraries and `ttnn`. This requires that all the libraries have the exact same compilation flags. 

`DEFINE_SYMBOL` by default would contain the name of the target, and so would be different for each sub-library, causing the PCH reuse to fail.